ksql-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -177,6 +198,30 @@ private KsqlRequest ksqlRequest(
     }
   }
 
+  private Future<Response> postAsync(
+      final String path,
+      final Object jsonEntity,
+      final Optional<Integer> readTimeoutMs
+  ) {
+    try {
+      // Performs an asynchronous request
+      return target
+          .path(path)
+          .request(MediaType.APPLICATION_JSON_TYPE)
+          .property(ClientProperties.READ_TIMEOUT, readTimeoutMs.orElse(0))
+          .headers(headers())
+          .async()
+          .post(Entity.json(jsonEntity));
+    } catch (final ProcessingException e) {
+      if (shouldRetry(readTimeoutMs, e)) {
+        postAsync(path, jsonEntity, calcReadTimeout(readTimeoutMs));
+      }
```

Comment:
it's usually better to retry in a loop instead of in the exception block. can you use a framework like `Retryer` or our `RetryUtil` to handle this?

Also even if it eventually succeeds, won't it throw because we still fall through...?

Reply:
I followed what the prior code did here